src/engine/shared/map.cpp

```diff
@@ -55,6 +56,7 @@ class CMap : public IEngineMap
 					
 					if(pTilemap->m_Version > 3)
 					{
+						dbg_assert((pTilemap->m_Width <= INT_MAX / 2 - 1) && (pTilemap->m_Height <= INT_MAX / 2 - 1), "map layer too big (%d * %d causes an integer overflow)");
```

Comment:
Would do an overflow check here and then abort the map loading in case it fails.
And while on it also check if the following mem_alloc returns 0 and abort then as well.